week06/manager_test.go

```diff
@@ -0,0 +1,211 @@
+package week06
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestManager_Start(t *testing.T) {
+	t.Parallel()
+
+	//create new manager
+	manager := NewManager()
```

Comment:
You should probably move this into the sub-test itself. You are starting and stopping a resource shared between your sub-tests.


It is always import to clean up your resources. Currently, if your test fails it will leak resources because the `m.Stop` function was never called to clean them up. You should immediately defer a call to `m.Stop` after creating your `Manager`.

```suggestion
	manager := NewManager()
	defer manager.Stop()
```